Session/Utilities/UIContextualAction+Utilities.swift

```diff
@@ -231,6 +231,32 @@ public extension UIContextualAction {
                             indexPath: indexPath,
                             tableView: tableView
                         ) { _, _, completionHandler in
+                            let isPinning: Bool = (threadViewModel.threadPinnedPriority <= 0)
+                            if isPinning,
+                               !dependencies[cache: .libSession].isSessionPro,
+                               let pinnedConversationsNumber: Int = dependencies[singleton: .storage].read({ db in
+                                   try SessionThread
+                                       .filter(SessionThread.Columns.pinnedPriority == 1)
```

Comment:
This should probably check `SessionThread.Columns.pinnedPriority >= 1` here - the `pinnedPriority` was designed in such a way so in the future it could be used to defined a fixed order for pinned conversations

If we were to _actually_ add that behaviour in a future release then this code would break because it's only handling the `== 1` case